### PR TITLE
Allow override of backend in dist.new_group()

### PR DESCRIPTION
**Pull request by pietern**
Summary:
There is no need to force the backend to be the same as the global
process group, as long as the backend is "nccl" or "gloo".

Differential Revision: D14657204

### CODE REVIEW

File: torch/distributed/distributed_c10d.py [hunk 1]
```diff
@@ -230,20 +230,28 @@ def _check_tensor_list(param, param_name):
...
     """
     return _NCCL_AVAILABLE
 
 
+def is_gloo_available():
+    """
+    Checks if the Gloo backend is available.
+
+    """
+    return True
```
Comment (apaszke) [inline torch/distributed/distributed_c10d.py]: This seems wrong? IIRC you can build PyTorch without gloo, so we should have a check for that

Comment (pietern) [reply torch/distributed/distributed_c10d.py]: @apaszke You can, but right now it won't build c10d, so it won't load `torch.distributed` at all. This should be fix in and by itself and then this function can be updated as well (added #18851).

**Comment by pietern** [comment]
@apaszke Updated `is_gloo_available` to do the right thing regardless of build configuration.

Added docstring for new `backend` kwarg to `new_group`.

**Comment by densechen** [comment]
Is there any demonstration of the usage of new_group()?

**Pull request closed without merging**